### ISSUE TITLE
[Proposal] Async setter for HTTPStore

### ISSUE THREAD

**Issue by manzt** [gzuidhof/zarr.js#12]
In #11 we add support for a concurrency-limited async getter. In a similar manner, we can add this functionality to `ZarrArray.set()`. 

I have this working locally, but think I should make a separate PR from #11. Once that is merged, I'll open another PR. This PR will also include a `SetOptions` interface which right now will be identical to `GetOptions` in #11. We could create a shared interface because they are identical currently, but perhaps in the future there will be options which are unique to gettting and setting.

```javascript
interface GetOptions {
  concurrencyLimit?: number;
  progressCallback?: (progressUpdate: {
    progress: number;
    queueSize: number;
  }) => void;
}

interface SetOptions {
  concurrencyLimit?: number;
  progressCallback?: (progressUpdate: {
    progress: number;
    queueSize: number;
  }) => void;
}
```

Cheers!

**Comment by manzt** [comment]
Related PR #14

**Comment by manzt** [comment]
Added with #14
